parquet/src/main/java/org/apache/iceberg/data/parquet/InternalReader.java

```diff
@@ -72,4 +91,12 @@ protected ParquetValueReader<?> timeReader(ColumnDescriptor desc) {
   protected ParquetValueReader<?> timestampReader(ColumnDescriptor desc, boolean isAdjustedToUTC) {
     return ParquetValueReaders.timestamps(desc);
   }
+
+  public void setCustomType(int fieldId, Class<? extends StructLike> type) {
```

Comment:
have this class implements the `SupportsCustomTypes` interface like Avro InternalReader